config/initializers/active_support_extensions.rb

```diff
@@ -0,0 +1,30 @@
+#this makes sure that hash with indifferent access gets deep converted into hash.
```

Comment:
What about putting here some condition on `Rails.version < 4`, maybe with some warning in else to make sure we remove this code once not needed. Also a comment about the origin of the code would be nice